recipes/wt/all/conanfile.py

```diff
@@ -185,56 +177,62 @@ def _gather_libs(p):
+            tc.variables["MYSQL_LIBRARIES"] = ";".join(_gather_libs("libmysqlclient"))
+            tc.variables["MYSQL_INCLUDE"] = ";".join(self.deps_cpp_info["libmysqlclient"].include_paths)
+            tc.variables["MYSQL_DEFINITIONS"] = ";".join("-D%s" % d for d in self.deps_cpp_info["libmysqlclient"].defines)
+            tc.variables["MYSQL_FOUND"] = True
         if self.options.get_safe("with_postgres"):
-            self._cmake.definitions["POSTGRES_PREFIX"] = self.deps_cpp_info["libpq"].rootpath
-            self._cmake.definitions["POSTGRES_LIBRARIES"] = ";".join(_gather_libs("libpq"))
-            self._cmake.definitions["POSTGRES_INCLUDE"] = ";".join(self.deps_cpp_info["libpq"].include_paths)
-            self._cmake.definitions["POSTGRES_FOUND"] = True
+            tc.variables["POSTGRES_PREFIX"] = self.deps_cpp_info["libpq"].rootpath
+            tc.variables["POSTGRES_LIBRARIES"] = ";".join(_gather_libs("libpq"))
+            tc.variables["POSTGRES_INCLUDE"] = ";".join(self.deps_cpp_info["libpq"].include_paths)
+            tc.variables["POSTGRES_FOUND"] = True
         if self.options.get_safe("with_mssql") and self.settings.os != "Windows":
-            self._cmake.definitions["ODBC_PREFIX"] = self.deps_cpp_info["odbc"].rootpath
-            self._cmake.definitions["ODBC_LIBRARIES"] = ";".join(_gather_libs("odbc"))
-            self._cmake.definitions["ODBC_INCLUDE"] = ";".join(self.deps_cpp_info["odbc"].include_paths)
-            self._cmake.definitions["ODBC_FOUND"] = True
+            tc.variables["ODBC_PREFIX"] = self.deps_cpp_info["odbc"].rootpath
+            tc.variables["ODBC_LIBRARIES"] = ";".join(_gather_libs("odbc"))
+            tc.variables["ODBC_INCLUDE"] = ";".join(self.deps_cpp_info["odbc"].include_paths)
+            tc.variables["ODBC_FOUND"] = True
         if self.options.get_safe("with_unwind"):
-            self._cmake.definitions["UNWIND_PREFIX"] = self.deps_cpp_info["libunwind"].rootpath
+            tc.variables["UNWIND_PREFIX"] = self.deps_cpp_info["libunwind"].rootpath
         if self.settings.os == "Windows":
-            self._cmake.definitions["CONNECTOR_FCGI"] = False
-            self._cmake.definitions["CONNECTOR_ISAPI"] = self.options.connector_isapi
+            tc.variables["CONNECTOR_FCGI"] = False
+            tc.variables["CONNECTOR_ISAPI"] = self.options.connector_isapi
         else:
-            self._cmake.definitions["CONNECTOR_FCGI"] = self.options.connector_fcgi
-            self._cmake.definitions["CONNECTOR_ISAPI"] = False
-        self._cmake.configure(build_folder=self._build_subfolder)
-        return self._cmake
+            tc.variables["CONNECTOR_FCGI"] = self.options.connector_fcgi
+            tc.variables["CONNECTOR_ISAPI"] = False
+        tc.generate()
+
+        cmake_deps = CMakeDeps(self)
+        cmake_deps.generate()
+
+    def layout(self):
+        cmake_layout(self, src_folder="source_folder")
```

Comment:
```suggestion
        cmake_layout(self, src_folder="src")
```

Reply:
Question: Is it preferred to specifying the src_folder or leave it out as a default argument?
Ive noticed most conan recipes leave it out eg 
```
cmake_layout(self)
```